src/Microsoft.AspNetCore.Razor/Parser/TagHelpers/Internal/TagHelperBlockRewriter.cs

```diff
@@ -207,15 +208,23 @@ public static class TagHelperBlockRewriter
                     // We're now at: " |asp-for='...'" or " |asp-for=..."
                     // The goal here is to capture the attribute name.
 
-                    var symbolContents = htmlSymbols
-                        .Skip(i) // Skip prefix
-                        .TakeWhile(nameSymbol => HtmlMarkupParser.IsValidAttributeNameSymbol(nameSymbol))
-                        .Select(nameSymbol => nameSymbol.Content);
+                    var nameSymbols = htmlSymbols.Skip(i); // Skip prefix
```

Comment:
Why not just make this a `for` loop and start at 1 - that will be simpler and faster than the current code :+1: